src/Tribe/REST/V1/Headers/Base.php

```diff
@@ -81,7 +81,36 @@ public function get_api_origin_meta_name() {
 	 */
 	public function get_rest_url() {
 		if ( is_single() && tribe_is_event() ) {
-			return tribe_events_rest_url( 'events/' . Tribe__Main::post_id_helper() );
+			$this_post_id = Tribe__Main::post_id_helper();
+
+			/**
+			 * We are dealing with an /all kind of query.
+			 * The root URL has to be modified to include the
+			 * post parent and its children post IDs.
+			 */
+			if (
+				( $wp_query = tribe_get_global_query_object() )
+				&& ( $post_parent = $wp_query->get( 'post_parent' ) )
+				&& $post_parent == $this_post_id
```

Comment:
Wondering about the loose comparison here instead of a strict equality.

Reply:
The query might provide the `post_parent` as a string or int. A cast before would be better was it not that it would throw on bad numeric values.
So to be able to `===` here an `is_numeric` check would need to be made before and then an `int` cast; in this case loose comparison is more efficient.